### DISCUSSION TITLE
Hubs not broadcasting and/or observing

### DISCUSSION THREAD

**Discussion by Cervesi** [pybricks/support#1926] [Q&A] [answered]
Hello, I have two Inventor hubs which refuse to broadcast and/or observe data. One and a half years ago they were working fine with the beta version, sending and receiving without issues (using Broadcast from pybricks.experimental at the time).

I recently tried to use them again with the most recent stable version but it seems that they broadcast no data. To test it, I wrote these two pieces of code, one per hub:

```
from pybricks.hubs import InventorHub
from pybricks.tools import wait

hub = InventorHub()
hub = InventorHub(broadcast_channel = 1)
hub = InventorHub(observe_channels = [2])

BDData = 1
while True:
    hub.ble.broadcast(BDData)
    ObsData = hub.ble.observe(2)
    if ObsData != None:
        hub.display.number(ObsData)
    wait(100)
```

```
from pybricks.hubs import InventorHub
from pybricks.tools import wait

hub = InventorHub()
hub = InventorHub(broadcast_channel = 2)
hub = InventorHub(observe_channels = [1])

BDData = 2
while True:
    hub.ble.broadcast(BDData)
    ObsData = hub.ble.observe(1)
    if ObsData != None:
        hub.display.number(ObsData)
    wait(100)

```
Normally I would expect to see a "2" displayed on hub 1 and an "1" on hub 2, instead the display never changes and the while cycle keeps running.
I tried reinstalling the firmware, kept them unplugged from the PC, but to no results. Using most recent stable version.

Thanks

**Comment by laurensvalk** [comment, marked as answer]
```python
hub = InventorHub()
hub = InventorHub(broadcast_channel = 1)
hub = InventorHub(observe_channels = [2])
```
Are these different tests or all in one script? This doesn't seem right.

You'd want just one hub initializer, like so:

```python
hub = InventorHub(broadcast_channel=1, observe_channels = [2])
```

And for the other hub:

```python
hub = InventorHub(broadcast_channel=2, observe_channels = [1])
```

Could you try again after updating as above?

**Reply by Cervesi** [reply]
Yes that is it! I thought I could stack multiple initializers. Thank you!

**Reply by laurensvalk** [reply]
Glad it worked!
